docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+      "type": "components",
+      "jsSrcsDir": "Libraries"
+    }
+  ]
+}
+```
+
+There's three arguments that are required:
+
+- `name`: A name of your library. This will be used to determine import path for your library.
+- `jsSrcsDir`: Path to the directory that contains the JavaScript specs for this library.
+
+These arguments are optional:
+
+- `type`: Optional. A string that determines which types of artifacts will be generated for your library: “modules” or “components”. If left unspecified, both modules and components artifacts will be generated.
+
+### 2. Extend or implement the code-generated native interfaces
+
+The JavaScript spec for your native module or component will be used to generate native interface code for each supported platform (i.e. Android and iOS). These native interface files will be generated when a React Native application that depends on your library is built.
+
+While this generated native interface code will not ship as part of your library, you do need to make sure your Objective-C or Java code conforms to the protocols provided by these native interface files. You can use the code-gen script to generate your library’s native interface code in order to use as reference. The files that are output by the script should not be committed, but you’ll need to refer to them to determine what changes you need to make to your native modules in order for them to provide an implementation for each generated @protocol / native interface.
+
+#### Conform to the protocols provided by the native interface code
+
+Update your native module or component to ensure it implements/extends the native interface that has been code-generated from your JavaScript specs.
+
+Following the example set forth in the previous section, your library might import MyAwesomeSpecs.h, extend the relevant native interface, and implement the necessary methods for this interface:
+
+```objc
+#import <MyAwesomeSpecs/MyAwesomeSpecs.h>
+
+@interface MyAwesomeModule () <StringGetterSpec>
+@end
+
+RCT_EXPORT_METHOD(getString:(NSString *)string
+                   callback:(RCTResponseSenderBlock)callback)
+{
+  // ...
+}
+```
+
+For an existing native module, you will likely already have one or more instances of `[RCT_EXPORT_METHOD](https://reactnative.dev/docs/native-modules-ios#export-a-native-method-to-javascript)`. To migrate to the new architecture, you’ll need to make sure the method signature makes use of the structs provided by the codegen output.
+
+# Supporting the New Architecture in Your Application
+
+## Prerequisites for Supporting the New Architecture in your Application
+
+There’s a few prerequisites that should be addressed before the new architecture is enabled in your application.
+
+### Use a React Native nightly release
+
+At this time, you must use a React Native nightly release in order to get access to the most up to date changes. Eventually, we will recommend targeting a minimum stable open source release.
```

Comment:
This guide has a lot of manual steps. I wonder if that's representative of what we can expect the final experience for existing apps to be like once Fabric is stable? If not, we should probably clarify that very prominently.

Reply:
> I wonder if that's representative of what we can expect the final experience for existing apps to be like once Fabric is stable? If not, we should probably clarify that very prominently.

Yup that's a great point. I've added a warning on top of the page + a warning in one of the Android section (as that felt the one where we require more manual steps). We can definitely make it more prominent.

We're also looking into automating some of the steps of the playbook as much as possible. To give an example this PR:
- https://github.com/facebook/react-native/pull/32790
Will allow users to enable all the new architecture machinery by turning a property on/off on Android for apps created with the new template. A similar PR has been landed for iOS as well.

Reply:
Added a banner on the intro page 👍 

![Screenshot 2022-01-06 at 17 30 06](https://user-images.githubusercontent.com/3001957/148424997-a269ba49-6357-41b9-a429-bfed1ff22926.png)